edx_notifications/stores/mongo/mongo_models.py

```diff
@@ -0,0 +1,25 @@
+"""
+Classes to support the Notification Store Mongodb backend
+"""
+from edx_notifications.base_data import DictField
+from edx_notifications.data import UserNotification
+from edx_notifications.stores.sql.models import SQLNotificationMessage
+
+
+class MongoUserNotification(object):
+
+    @classmethod
+    def to_data_object(cls, mongo_user_notification, user_id):
+        """
+        Generate a NotificationType data object
+        """
+
+        msg = SQLNotificationMessage.objects.get(id=mongo_user_notification['msg_id'])
```

Comment:
Yea, we should talk about this line as it will not be performant. I'll try to make some concrete suggestions.